### PR TITLE
fix: view origin is not 0,should be navigationbar height

### PR DESCRIPTION
**Pull request by godgnay**
fix: view origin is not 0,should be navigationbar height

# Description

When push to ViewController, TextField beignFirst。 The topViewBeginOrigin‘s value is {0,0,320,506}, I has to check version 5.0.0 ,find some code to fix origin value and copied. thanks!

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by hackiftekhar** [comment]
@godgnay, can you please first share demo project of issue?

**Comment by antonyalkmim** [comment]
Hello @hackiftekhar , I think this PR fix my problem. When I push some view controller on navigationController, its showing a black space at the bottom because my view controller top is moved to `top=0`where it should be `top=navigationBarHeight`.

#### Screenshot

**Comment by godgnay** [comment]
@hackiftekhar hello，the [demo](https://github.com/godgnay/IQKeyboardManagerDemo)，the [gif](https://github.com/godgnay/IQKeyboardManagerDemo/blob/master/demo.gif)，the fix code is my coped from your old version code，I just want tell you this bug show，ths

**Comment by godgnay** [comment]
demo

File: IQKeyboardManager/IQKeyboardManager.m [hunk 1]
```diff
@@ -1083,6 +1083,12 @@ -(void)keyboardWillShow:(NSNotification*)aNotification
         UIViewController *rootController = [textFieldView parentContainerViewController];
         _rootViewController = rootController;
         _topViewBeginOrigin = rootController.view.frame.origin;
+        if (([_rootViewController isKindOfClass:[UINavigationController class]] || _rootViewController.navigationController) &&
+            [_rootViewController modalPresentationStyle] != UIModalPresentationFormSheet &&
+            [_rootViewController modalPresentationStyle] != UIModalPresentationPageSheet)
+        {
+            _topViewBeginOrigin = CGPointMake(0, [self keyWindow].frame.size.height-_rootViewController.view.frame.size.height);
```
Comment (hackiftekhar) [inline IQKeyboardManager/IQKeyboardManager.m]: This will create issues if ViewController is added as childViewController

**Comment by hackiftekhar** [comment]
You are trying to show keyboard from viewDidLoad where the view might not be loaded fully. Please try to move becomeFirstResponse related code to viewDidAppear where view loaded fully.
```
-(void)viewDidAppear:(BOOL)animated
{
    [super viewDidAppear:animated];

    if (self.mShowKeyboard) {
        self.mShowKeyboard = NO;
        [self.mIDTextField becomeFirstResponder];
    }
}
```

**Pull request closed without merging**